### DISCUSSION TITLE
Tree: double click support

### DISCUSSION THREAD

**Discussion by jasdom** [iTwin/iTwinUI#2085] [Ideas]
Hi, we are replacing our tree components using `@itwin/itwinui-react`, and some of the trees require additional functionality when a node is double clicked. Are there any plans to allow passing a callback that is called when a node is double clicked?

**Comment by mayank99** [comment]
You can implement this yourself in user-land today by passing `onDoubleClick` via `<TreeNode nodeProps={{ onDoubleClick: () => {} }} />`. (Make sure you're on the latest version if you're not seeing this prop)

That being said, double-click behavior is non-standard and can lead to a poor user experience, especially if not using a mouse. Make sure to provide an alternative path for the user to perform the same action. Double-click should be seen as an optional shortcut.
